File: Back/PWEB2/src/app/shared/model/produto.ts
export class Produto {

  id?: string;
  nome = '';
  estoque: number;
  preco: number;

  constructor(id= "", produto: Produto = { nome: '', estoque: 0, preco: 0 }) {
    this.id = id;
    this.nome = produto.nome;
    this.estoque = produto.estoque;
    this.preco = produto.preco;
  }
}

File: Back/api (1)/api/src/main/java/br/edu/ifpb/gugawag/api/Modelo/Produto.java
package br.edu.ifpb.gugawag.api.Modelo;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import org.hibernate.id.factory.internal.AutoGenerationTypeStrategy;

import javax.annotation.processing.Generated;

@Entity
public class Produto {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;
    private String nome;
    private int estoque;
    private int preco;

    public Long getId(){return id;}

    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public int getEstoque() {
        return estoque;
    }

    public void setEstoque(int estoque) {
        this.estoque = estoque;
    }

    public int getPreco() {
        return preco;
    }

    public void setPreco(int preco) {
        this.preco = preco;
    }
}
